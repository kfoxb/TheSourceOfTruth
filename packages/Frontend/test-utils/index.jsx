/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
