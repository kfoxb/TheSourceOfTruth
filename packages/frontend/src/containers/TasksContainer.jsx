import React, { Component, Fragment } from 'react';
import { database } from 'firebase';
import { fromJS, List, Map } from 'immutable';
import { Link } from 'react-router-dom';
import { CREATE, DOCUMENTS, EDIT, PHASE, VIEW } from '@the-source-of-truth/shared/constants';
import Loading from '../components/Loading';

export default class TasksContainer extends Component {
  static createDocumentLinks(doc) {
    const title = doc.get('title');
    const phase = doc.get(PHASE);
    const viewHref = `/${DOCUMENTS}/${VIEW}/${doc.get('id')}`;
    const editHref = `/${DOCUMENTS}/${phase}/${doc.get('id')}`;
    return (
      <div key={doc.get('id')} >
        <Link
          to={viewHref}
          href={viewHref}
        >
          {title || 'Untitled'}
        </Link>
        <Link
          to={editHref}
          href={editHref}
        >
          {'    (continue)'}
        </Link>
      </div>
    );
  }

  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      documents: fromJS({
        [CREATE]: [],
        [EDIT]: [],
      }),
    };
  }

  componentDidMount() {
    this.ref = database()
      .ref(DOCUMENTS);
    this.setListenerByPhase(CREATE);
    this.setListenerByPhase(EDIT);
  }

  componentWillUnmount() {
    this.ref.off();
  }

  setListenerByPhase(phase) {
    this.ref
      .orderByChild(PHASE)
      .equalTo(phase)
      .on('value', this.handleSnapshot(phase), this.handleError);
  }

  handleError = (error) => {
    // eslint-disable-next-line no-console
    console.error(error);
  }

  handleSnapshot = phase => (snapshot) => {
    if (this.state.loading) {
      this.setState({ loading: false });
    }
    const data = snapshot.val();

    const documents = Object.keys(data)
      .reduce((cur, acc) => cur.push(new Map({ id: acc, ...data[acc] })), new List([]));

    this.setState({ documents: this.state.documents.set(phase, documents) });
  }

  render() {
    if (this.state.loading) {
      return (<Loading />);
    }
    return (
      <Fragment>
        <button><a href={`/${DOCUMENTS}/create`}>Create Document</a></button>
        <h3>Create Tasks</h3>
        { this.state.documents.get(CREATE).map(TasksContainer.createDocumentLinks) }
        <h3>Edit Tasks</h3>
        { this.state.documents.get(EDIT).map(TasksContainer.createDocumentLinks) }
      </Fragment>
    );
  }
}
