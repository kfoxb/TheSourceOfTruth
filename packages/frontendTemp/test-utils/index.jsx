/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
