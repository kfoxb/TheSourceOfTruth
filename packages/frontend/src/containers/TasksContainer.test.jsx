/* eslint-env jest */
import { createAssertWithPropsToMatchSnapshot } from '../../test-utils';
import TasksContainer from './TasksContainer';

jest.mock('firebase');

describe('TasksContainer', () => {
  const defaultProps = {};

  const assertWithPropsToMatchSnapshot =
    createAssertWithPropsToMatchSnapshot(TasksContainer.WrappedComponent, defaultProps);
  it('should render', () => {
    assertWithPropsToMatchSnapshot();
  });
});
