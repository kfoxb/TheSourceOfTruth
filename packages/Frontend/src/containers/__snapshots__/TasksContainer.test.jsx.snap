// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksContainer should render 1`] = `<WithStyles(LinearIndeterminate) />`;
